### ISSUE TITLE
Improve API paging behavior

### ISSUE THREAD

**Issue by ppratscher** [prysmaticlabs/prysm#4174]
Currently paged API calls are a little bit difficulty to handle. Generally all paged API calls return a next page token even if there is no next page. If you pass a next page token that returns no results the api throws an error:
```
rpc error: code = Internal desc = Could not paginate results: page start 1000 >= list 501
```

Instead it would be much nicer for the api to return an empty next page token in case there are no further pages available. If the caller asks for a non-existing page the api should return an empty response instead of an error.

Both changes would allow a much easier handling of paged api calls.

**Comment by prestonvanloon** [comment]
@rauljordan please triage as necessary

**Comment by rauljordan** [comment]
Closed by #4176
